### PR TITLE
switch to psr-4

### PR DESCRIPTION
**Pull request by garak**
This will fix issue #227 
Also, allow php 7.4 to fail (not due to this library, doctrine annotations are to blame)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -47,13 +47,13 @@
...
-            "Knp\\Component": "src"
+        "psr-4": {
+            "Knp\\Component\\": "src/Knp/Component"
         }
     },
     "autoload-dev": {
-        "psr-0": {
-            "Test": "tests"
+        "psr-4": {
+            "Test\\": "tests/Test"
```
Comment (polc) [inline composer.json]: Maybe `Knp\Component\Tests`?

Comment (polc) [reply composer.json]: It's how it's done in Symfony components for example https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/EventDispatcher/Tests/EventDispatcherTest.php#L12

Comment (garak) [reply composer.json]: That is the old namespace and I kept it just to minimize changes. If you think it's important, I can change it.

Comment (polc) [reply composer.json]: oh okay, not important :+1:

**Pull request merged**